packages/build-utils/src/clone-env.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Performs a shallow clone on the environment variable object.
+ *
+ * @param {Object} [currentEnv] The environment object to clone
+ * @returns {Object}
+ */
+export function cloneEnv(currentEnv = process.env): typeof process.env {
+  return {
+    // the system path is called `Path` on Windows and Node.js will
+    // automatically return the system path when accessing `PATH`,
+    // however we lose this proxied value when we destructure and
+    // thus we must explicitly copy it
+    PATH: currentEnv.PATH,
+    ...currentEnv,
```

Comment:
Should PATH be moved to after instead of before?

What happens when you want to override the PATH after cloning such as here:

https://github.com/vercel/vercel/blob/177835b58864e6750e10c590721f487c0b0f73a5/packages/build-utils/src/fs/run-user-scripts.ts#L241

Reply:
I don't believe there is a difference whether `PATH` is explicitly set before or after the destructure of the current environment variable object. I think it just comes down to a coding style preference and/or readability.

Not sure I follow you with the second question. `opts.env` is a new object that is a shallow clone of `process.env` with `Path` copied to `PATH`. It is possible for `opts.env` on Windows to have both a `Path` and `PATH` with identical values. If later you want to override `opts.env.PATH`, it will only mutate the cloned object.